MeshLib/NodeAdjacencyTable.h

```diff
@@ -0,0 +1,80 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2014, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MESHLIB_NODE_ADJACENCE_TABLE_H_
+#define MESHLIB_NODE_ADJACENCE_TABLE_H_
+
+#include <algorithm>
+#include <vector>
+
+#include "Node.h"
+
+
+namespace MeshLib
+{
+
+/// Representation of topological node adjacency.
+///
+/// The topological sparsity pattern in the context of FEM is defined in terms of
+/// supports of the nodal functions. Especially, two nodes i and j are called
+/// adjacent if and only if there is a mesh element E including nodes i and j.
+/// This information is represented by the NodeAdjacenceTable.
+///
+/// The topological adjacency of nodes is created by
+/// Mesh::setNodesConnectedByElements() which is usually called upon mesh
+/// construction.
+class
+NodeAdjacencyTable
+{
+
+public:
+    NodeAdjacencyTable() = default;
+
+    explicit
+    NodeAdjacencyTable(std::vector<Node*> const& nodes)
+    {
+        _data.resize(nodes.size());
+
+        createTable(nodes);
+    }
+
+    std::size_t size() const
+    {
+        return _data.size();
+    }
+
+    std::size_t getNodeDegree(std::size_t const node_id) const
+    {
+        return _data[node_id].size();
+    }
+
+    void createTable(std::vector<Node*> const& nodes)
+    {
```

Comment:
MeshComponentMap should be considered here or somewhere in the LIS solver initialization if a problem has more than one unknowns.